### PR TITLE
Tim/dummy rule

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
I have a feeling I don't have all cases covered, but I fail to come up with any that don't work.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/Rules/TimFirstRule.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                if (c.Node is ForEachStatementSyntax node)
+                {
+                    ReportIfIdentifierIsNotTimLowerCase(c, node.Identifier);
+                }
+            }, SyntaxKind.ForEachStatement);
+        // Part 2
+        context.RegisterNodeAction(c =>
+            {
+                if (c.Node is IdentifierNameSyntax node
+                    && c.SemanticModel.GetSymbolInfo(node).Symbol is { Kind: SymbolKind.Property } symbol)
```
Comment (MaStr11) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/TimFirstRule.cs:60]: That's the way to do it. It requires some more test with e.g. namespaces where IdentifierName can be found. You may want to have a look in the grammar to find more examples.

**Review by MaStr11 — COMMENTED** [review]
I left you commemts as I would in a first round (no naming review, no code style, just general hints about missing cases or other ways to do something). There is no real point in addressing any of the comments as long as you know what was meant by the comment. Feel free to ask or comment if there are questions and address whatever you think might be interesting to solve.

File: analyzers/src/SonarAnalyzer.CSharp/Rules/TimFirstRule.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        context.RegisterNodeAction(c =>
+            {
+                if (c.Node is LocalDeclarationStatementSyntax { Declaration.Variables: var variables } node)
+                {
+                    foreach (var variable in variables)
+                    {
+                        ReportIfIdentifierIsNotTimLowerCase(c, variable.Identifier);
+                    }
+                }
+            }, SyntaxKind.LocalDeclarationStatement);
```
[the remark this answers is not in the document]
Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/TimFirstRule.cs:45]: This is much nice and also caught a case, I missed (for-loop), thanks. I cannot find any FPs, though.

**Comment by Tim-Pohlmann** [comment]
@martin-strecker-sonarsource what to do with this PR? Close it?

**Pull request closed without merging**